lib/dns.js

```diff
@@ -189,6 +189,9 @@ exports.lookupService = function(host, port, callback) {
   if (cares.isIP(host) === 0)
     throw new TypeError('"host" argument needs to be a valid IP address');
 
+  if (typeof port !== 'number')
```

Comment:
I agree with accepting a string for the port for consistency, but will that make this semver-major since it is a behavior change?